org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/ChangeConverter.xtend

```diff
@@ -31,21 +35,35 @@ class ChangeConverter implements IAcceptor<IEmfResourceChange> {
 
 	static class Factory {
 
-		@Inject IResourceServiceProvider.Registry registry
+		@Inject protected IResourceServiceProvider.Registry registry
 
+		/**
+		 * @deprecated use {@link #create(WorkspaceManager, WorkspaceEdit, boolean}�instead.
+		 */
+		@Deprecated
 		def ChangeConverter create(WorkspaceManager workspaceManager, WorkspaceEdit edit) {
-			new ChangeConverter(workspaceManager, registry, edit)
+			new ChangeConverter(workspaceManager, registry, edit, null)
+		}
+
+		def ChangeConverter create(WorkspaceManager workspaceManager, WorkspaceEdit edit, IRenameServiceExtension.Options options) {
+			new ChangeConverter(workspaceManager, registry, edit, options)
 		}
 	}
 
 	val WorkspaceManager workspaceManager
 	val IResourceServiceProvider.Registry registry
 	val WorkspaceEdit edit
+	val IRenameServiceExtension.Options options
 	
-	protected new(WorkspaceManager workspaceManager, IResourceServiceProvider.Registry registry, WorkspaceEdit edit) {
+	protected new(WorkspaceManager workspaceManager, IResourceServiceProvider.Registry registry, WorkspaceEdit edit, IRenameServiceExtension.Options options) {
```

Comment:
Could it be that clients may have specialized the ChangeConverter and the constructor still would need to callable as is.

Reply:
Yes, that's where I stopped to be API compatible cause I think this is highly unlikely.